### ISSUE TITLE
Implementing Player-Zero Training (2/2)

### ISSUE THREAD

**Issue by Klazkin** [Klazkin/player-zero#83]
Explain the context of the issue, what is being addressed in detail.

## The goal

- [ ] Board Input normalization
- [ ] Look into weight standardization https://arxiv.org/pdf/1903.10520v1.pdf
- [ ] ConvNext https://github.com/facebookresearch/ConvNeXt
- [x] Data generation during tournament
- [x] BUG same node sub-tree has actions for different casters
- [x] Single training instance
- [x] Running training server
- [x] increase number of simulations due to large branch factor
- [x] verify that winner is relative to current unit
- [x] policy outputs should be softmax'ed'
- [x] normalize the node scores, normalize policy outputs
- ~~normalize value~~ unnecessary?
- [x] add mask to policy outputs
- [x] Implement removal of old files
- [ ] Configure optimizer
- [x] Configure Regularization
- [x] Configure loading of old model weights for training
- [ ] There is probably no reason to constantly save and load models if the training server is running, but only save for checkpoints
- [x] #84 

* BLOODDRAWING child scores do not account for the possible random outcomes - currently score is calculated only once for one of the two outcomes, but realistically it should be done twice (or more, depending on number of branches).

## Time tracking

Time Estimate: `undefined`
Time spent: `7 hours 48 minutes`

## Resources
https://realpython.com/python-sockets/
https://docs.godotengine.org/en/stable/classes/class_packedbytearray.html
https://docs.godotengine.org/en/stable/classes/class_streampeertcp.html#class-streampeertcp
https://docs.godotengine.org/en/stable/classes/class_streampeer.html#class-streampeer-method-get-string
https://keras.io/guides/serialization_and_saving/
https://rl-vs.github.io/rlvs2021/class-material/regularized_mdp/Regularization_RL_RLVS.pdf
